manifests/baremetal/coredns-corefile.tmpl

```diff
@@ -10,9 +10,9 @@
         {{ .ControllerConfig.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP }} api.{{ .ControllerConfig.EtcdDiscoveryDomain }}
         fallthrough
     }
-    template IN A {{ .ControllerConfig.EtcdDiscoveryDomain }} {
+    template IN ANY {{ .ControllerConfig.EtcdDiscoveryDomain }} {
         match .*.apps.{{ .ControllerConfig.EtcdDiscoveryDomain }}
-        answer "{{`{{"{{ .Name }}"}}`}} 60 in a {{ .ControllerConfig.Infra.Status.PlatformStatus.BareMetal.IngressIP }}"
+        answer "{{`{{"{{ .Name }}"}}`}} 60 in {{`{{"{{ .Type }}"}}`}} {{ .ControllerConfig.Infra.Status.PlatformStatus.BareMetal.IngressIP }}"
```

Comment:
Will using ANY and .Type here not mean that when we get an A request, we will answer with A but with an IPv6 address?

Reply:
Yes, but at least when we get the right type of request we'll return the right type. Making this return an AAAA record when an A record was requested is going to be a lot trickier. We'd have to specify the type based on the ip version of the ingress VIP, and I don't think the templating language has a way to do that built in. We'd have to write an external function that did the version detection.

Reply:
We should use runtimecfg to detect the IP family we are using for the VIPs and set the record type accordingly.